Library/Homebrew/requirements/macos_requirement.rb

```diff
@@ -0,0 +1,26 @@
+require "requirement"
+
+class MacOSRequirement < Requirement
+  fatal true
+
+  def initialize(tags = [])
+    @version = MacOS::Version.from_symbol tags.first unless tags.empty?
+    super
+  end
+
+  satisfy(build_env: false) do
+    next false if !OS.mac? && !@version
+    next true if !OS.mac? || !@version
+    MacOS.version >= @version
+  end
+
+  def message
+    return "macOS is required." if !OS.mac? || !@version
```

Comment:
`if !OS.mac? || !@version` could be a utility method, I think, it's repeated 3 times.